src/controllers/playback/videoosd.js

```diff
@@ -1088,18 +1088,33 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
         var NavigationKeys = ["ArrowLeft", "ArrowRight", "ArrowUp", "ArrowDown"];
 
         function onWindowKeyDown(e) {
```

Comment:
Have you checked for any other usages of `onWindowKeyDown` for other purposes?

Reply:
In Chrome most of cases (keyboard and one pair of gamepad) works. "Left"/"Right" and other gamepad alternatives are remain untested. But `getKeyName` returns the same `e.key` for buttons that are not listed in the array, and `onWindowKeyDown` for them remained the same.

I did not check Firefox. Now I checked, keyboard works, except for `Space`. It looks like the videoplayer receives a `click` from `Space` and pauses. Then videoosd catches `Space` and unpauses. Tried on clean JF - same behavior.
https://github.com/jellyfin/jellyfin-web/blob/ec6ce5aa14384f51ff4f97a749bb4e0985aed773/src/controllers/playback/videoosd.js#L1093-L1097

Reply:
Can you add a browser check there so we can stop the behavior then? Maybe with a comment explaining why.

Reply:
I looked again and found a way to repeat similar to this in Chrome.
* We at the state when OSD is hidden.
* Press and hold `Space`.
* `videoosd` pauses player and shows OSD.
* Focus moves to `Play/Pause` button.
* Release `Space` - this generates `click` on `Play/Pause` button, and playback resumes.

It seems that in Firefox, hidden button still has focus (CSS styles `:focus` while hidden). That's why it is more "sensitive", I think.

I tried to add `document.activeElement.blur()` in `hideMainOsdControl` - Firefox started working similarly to Chrome (I notice how the focus goes away). But an issue with `Space` holding still remains in both.

Reply:
Previous solution was still unstable.
<details>
<summary>The new one</summary>

```diff
diff --git a/src/controllers/playback/videoosd.js b/src/controllers/playback/videoosd.js
index 6392c3a1a..f192abbb5 100644
--- a/src/controllers/playback/videoosd.js
+++ b/src/controllers/playback/videoosd.js
@@ -437,6 +437,11 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
                 });
                 currentVisibleMenu = null;
                 toggleSubtitleSync("hide");
+
+                // Firefox does not blur by itself
+                if (document.activeElement) {
+                    document.activeElement.blur();
+                }
             }
         }
 
@@ -1087,7 +1092,15 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
          */
         var NavigationKeys = ["ArrowLeft", "ArrowRight", "ArrowUp", "ArrowDown"];
 
+        /**
+         * Clicked element.
+         * To skip `click` handling on Firefox/Edge.
+         */
+        var clickedElement;
+
         function onWindowKeyDown(e) {
+            clickedElement = e.srcElement;
+
             var key = keyboardnavigation.getKeyName(e);
 
             if (!currentVisibleMenu && 32 === e.keyCode) {
@@ -1159,6 +1172,14 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
             }
         }
 
+        function onWindowMouseDown(e) {
+            clickedElement = e.srcElement;
+        }
+
+        function onWindowTouchStart(e) {
+            clickedElement = e.srcElement;
+        }
+
         function getImgUrl(item, chapter, index, maxWidth, apiClient) {
             if (chapter.ImageTag) {
                 return apiClient.getScaledImageUrl(item.Id, {
@@ -1289,6 +1310,12 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
                 dom.addEventListener(window, "keydown", onWindowKeyDown, {
                     capture: true
                 });
+                dom.addEventListener(window, window.PointerEvent ? "pointerdown" : "mousedown", onWindowMouseDown, {
+                    passive: true
+                });
+                dom.addEventListener(window, "touchstart", onWindowTouchStart, {
+                    passive: true
+                });
             } catch (e) {
                 require(['appRouter'], function(appRouter) {
                     appRouter.showDirect('/');
@@ -1303,6 +1330,12 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
             dom.removeEventListener(window, "keydown", onWindowKeyDown, {
                 capture: true
             });
+            dom.removeEventListener(window, window.PointerEvent ? "pointerdown" : "mousedown", onWindowMouseDown, {
+                passive: true
+            });
+            dom.removeEventListener(window, "touchstart", onWindowTouchStart, {
+                passive: true
+            });
             stopOsdHideTimer();
             headerElement.classList.remove("osdHeader");
             headerElement.classList.remove("osdHeader-hidden");
@@ -1472,7 +1505,9 @@ define(["playbackManager", "dom", "inputManager", "datetime", "itemHelper", "med
             playbackManager.previousTrack(currentPlayer);
         });
         view.querySelector(".btnPause").addEventListener("click", function () {
-            playbackManager.playPause(currentPlayer);
+            if (this.contains(clickedElement)) {
+                playbackManager.playPause(currentPlayer);
+            }
         });
         view.querySelector(".btnNextTrack").addEventListener("click", function () {
             playbackManager.nextTrack(currentPlayer);
```
</details>

The idea is to skip `click` if keydown/mousedown/pointerdown/touchstart was on different element.

Reply:
That looks good to me, can you apply it here?